Adopt the Plugin Index RFC's format

MCOfficer opened the pull request:
This PR makes the repository compliant with the plugin index format proposed in [this RFC](https://github.com/EndlessSkyCommunity/endless-sky/blob/community/docs/rfcs/text/0001-plugin-index.md).

Note that the link to PLUGINS.md in the README still points at the master branch, not at the version from this PR.

As of now, there are a couple question marks for me:

1. Should I (try) to convert all plugins to manifests, or leave it to the plugin's authors? I lean towards the latter, because with the format changes have come a few fields that I have to infer. It would be better to have plugin authors add their plugins, as that's how it's supposed to be anyways.
Tagging @Darcman99 @RestingImmortal @Anarchist2 as you are somewhat involved in this and have plugins at stake.
2. Are we okay with having filenames like this? ("Ciu Kreo.yaml")

Edit: also, mandatory @tehhowch ping. Sorry mate, but you always have input.

tehhowch commented on scripts/generate_table.py:
```diff
@@ -10,21 +10,21 @@
-        .replace("\n", " ") \
-        .replace("\r", "") \
-        .strip()
+    plugin["description"] = (
+        plugin["description"].replace("\n", " ").replace("\r", "").strip()
+    )
```
Should we even display the description in the md table? That can be something for the individual md files, where we don't have to worry about wildly inconsistent row heights. Users will still have the short description.

MCOfficer replied:
Good call, i just skimmed over this. We should use shortDescription instead.

tehhowch commented on manifests/All Content Plugin.yaml:
```diff
@@ -0,0 +1,16 @@
+name: High DPI
```
```suggestion
name: All Content Plugin
```

MCOfficer replied:
oops.

tehhowch reviewed (commented):
1) it's fine to have authors come update their stuff.
2) What's the concern? Spaces in file names?

MCOfficer commented on scripts/autoupdate.py:
```diff
@@ -0,0 +1,127 @@
+def get_latest_version(au_type, au_url, au_branch):
+    if au_type == "commit":
+        refs = ls_remote(au_url)
+        ref = refs.get("refs/heads/" + au_branch)
```
the `[]` accessor throws the KeyError, `get` returns null ([ref](https://stackoverflow.com/questions/11041405/why-dict-getkey-instead-of-dictkey)).

MCOfficer commented on scripts/autoupdate.py:
```diff
@@ -0,0 +1,127 @@
+        key: 0 if value is None else value for key, value in match.groupdict().items()
+    }
+    ver = semver.VersionInfo(**ver)
+    rest = match.string[match.end() :]  # noqa:E203
```
You're correct, but I don't use that return value anyways. I opted not to modify this function so i don't have to retest it, instead i can just trust the python-semver devs to know what they're doing.

MCOfficer commented on scripts/autoupdate.py:
```diff
@@ -0,0 +1,127 @@
+                versions.append((tag, coerce(tag)))
+
+        # Sort by the second tuple item, which is a parser semver object
+        # Then return the associated string (so any `v` prefix that `coerce` stripped doesn't get discarded)
```
Let's just get rid of the \`v\` for clarity.

MCOfficer commented on scripts/check_urls.py:
```diff
@@ -1,46 +1,51 @@
-    print("Testing " + url)
+def check_url(url):
+    print("  Testing " + url)
     try:
         conn = urlopen(Request(url, method="HEAD"))
         return True
     except HTTPError as e:
-        print("%s: %s" % (e.code, e.reason))
+        print("  %s: %s" % (e.code, e.reason))
```
I don't think i understand. Are you referring to the indentation, or the error formatting?

tehhowch replied:
There are two prints. Let's just have one print that has the information from both.

MCOfficer replied:
To reduce the output, or the code size? Cause i can't think of a way to do this without more code.

tehhowch replied:
I suggested it so the success or failed state can be determined from one line instead of "maybe two"

An alternative would be for this method to return None on success, and the error message on error. Then the caller can log it, if they care to.

MCOfficer replied:
How's this?

```py
def check_url(url):
    try:
        conn = urlopen(Request(url, method="HEAD"))
        status = "OK %s" % conn.status
        ok = False
    except HTTPError as e:
        status = "ERROR %s" % e.code
        ok = True
    
    print("  %s\t%s" % (status, url))
    return ok
```

```
Checking manifests/Ciu Kreo.yaml
  OK 200        https://github.com/RestingImmortal/Ciu-Kreo/archive/c8deb9aee5c7e0b96657d21e76b81602acbcb8a1.zip
  OK 200        https://github.com/RestingImmortal/Ciu-Kreo/raw/c8deb9aee5c7e0b96657d21e76b81602acbcb8a1/icon.png
Checking manifests/All Content Plugin.yaml
  ERROR 404     https://github.com/endless-sky/all-content-plugin/archive/refs/tasfdags/v0.9.13.zip
  OK 200        https://github.com/endless-sky/all-content-plugin/raw/v0.9.13/icon.png

```

tehhowch replied:
Sure (if your `ok` are meant to be like that). My concern was that grepping for a particular plugin wouldn't always get you the full context.

MCOfficer commented:
Sorry, forgot to answer to review while addressing the comments.

>     2. What's the concern? Spaces in file names?

Nothing specific, most software repos just prefer kebab case. I don't see any downside to this until we allow special characters (`Becca's` instead of `Beccas`), then URLs could get weird.

tehhowch commented:
If we need to add a url validator / linter step in the review to ensure we always catch that, we should. I definitely foresee folks authoring plugins where their preferred plugin title contains characters that require url encoding, i.e. our index implementation shouldn't prevent the reasonable use of characters by authors.

MCOfficer commented:
> 
> 
> If we need to add a url validator / linter step in the review to ensure we always catch that, we should. I definitely foresee folks authoring plugins where their preferred plugin title contains characters that require url encoding, i.e. our index implementation shouldn't prevent the reasonable use of characters by authors.

Honestly, before we follow convention for no reason, we should think about whether it makes sense. I don't know the exact reason why linux repos prefer kebab case, but i reckon it's for simplicity towards humans. Since this index is meant to be machine-readable (users only see it via a GUI), and programs should be able to url-encode in this day and age, special characters don't seem like a problem to me.
